### PR TITLE
Assertion fix in RegExp.

### PR DESCRIPTION
**Pull request by LaszloLango**
Assertion 'mem_pools == NULL' failed in mem_pools_finalize

JerryScript-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by LaszloLango** [comment]
Related issue #255

**Comment by egavrin** [comment]
Good to me

File: jerry-core/ecma/operations/ecma-regexp-object.cpp [hunk 1]
```diff
@@ -1003,12 +1003,13 @@ re_match_regexp (re_matcher_ctx_t *re_ctx_p, /**< RegExp matcher context */
           ecma_completion_value_t match_value = re_match_regexp (re_ctx_p, bc_p, str_p, &sub_str_p);
           if (!ecma_is_value_true (match_value))
```
Comment (galpeter) [inline jerry-core/ecma/operations/ecma-regexp-object.cpp]: Not related to this PR:
The `ecma_is_value_true` call is not correct for the `match_value` as the function expects `ecma_value_t`, but the variable is an `ecma_completion_value_t`.

Comment (zherczeg) [inline jerry-core/ecma/operations/ecma-regexp-object.cpp]: It is sad that the compiler didn't catch this. We should make a fix for this.

**Comment by zherczeg** [comment]
+1 lgtm

**Comment by egavrin** [comment]
`make push`

File: jerry-core/ecma/operations/ecma-regexp-object.cpp (hunk 1 above)
Comment (ruben-ayrapetyan) [inline jerry-core/ecma/operations/ecma-regexp-object.cpp]: @zherczeg, please, see #139.

**Pull request closed without merging**